krm-functions/lib/nad/v1/nad_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2023 Nephio.
+Copyright 2023 Samsung.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"fmt"
+	"testing"
+
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestParseKubeObjectByNADInterface(t *testing.T) {
```

Comment:
Maybe a good idea to come the two different test functions into one single `TestParseKubeObject` using table/map driven tests